src/Dnn.PersonaBar.Library/DTO/PersistSettings.cs

```diff
@@ -0,0 +1,89 @@
+﻿#region Copyright
+// 
+// DotNetNuke® - http://www.dotnetnuke.com
+// Copyright (c) 2002-2018
+// by DotNetNuke Corporation
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
+// of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
+// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+// DEALINGS IN THE SOFTWARE.
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Xml;
+using System.Xml.Schema;
+using System.Xml.Serialization;
+using DotNetNuke.Common.Utilities;
+
+namespace Dnn.PersonaBar.Library.DTO
+{
+    /// <summary>
+    /// Persona Bar Persist Settings For User.
+    /// </summary>
+    [DataContract]
+    public class PersistSettings : Dictionary<string, object>, IXmlSerializable
```

Comment:
This is not a good name. It almost suggests that the current`UserSettings` class is not being persisted. Maybe it would be possible to extend instead with the ability to store any arbitrary, custom settings?

Reply:
hmm, doesn't have good name as UserSettings still need keep in class for backward compatible. CustomSettings not good neither as this is user-level settings.